web/src/app/components/cdf/cdf.component.ts

```diff
@@ -250,22 +262,275 @@ export class CDFComponent implements AfterViewInit {
 
-    addCurrentPushLine(
-        this.currentPush, currentPushLine, this.data, height, xScale, yScale);
+    cdfChart.append('text')
+        .attr('x', xScale(startValue) - 35)
+        .attr(
+            'y',
+            yScale(getProbabilityForDuration(this.data, startValue) / 100) + 30)
+        .attr('class', 'click-line-y-text')
+        .attr('font-size', '10px')
+        .text(`${this.data.filter(c => c.duration <= startValue).length}/${
+            this.data.length}`)
+        .attr('opacity', 0);
+
+    cdfChart.append('text')
+        .attr('x', xScale(startValue) / 2)
+        .attr(
+            'y',
+            yScale(getProbabilityForDuration(this.data, startValue) / 100) + 10)
+        .attr('class', 'click-line-x-text')
+        .attr('font-size', '10px')
+        .text(`${getProbabilityForDuration(this.data, startValue).toFixed(2)}%`)
+        .attr('opacity', 0);
+
+    cdfChart.on('click', (d: unknown, i: number): void => {
+      if (!d) {
+        return;
+      }
+      const coordinates = d3.mouse(d3.event.currentTarget);
+      const xValue = xScale.invert(coordinates[0]);
+      if (xValue > minDuration) {
+        startValue = xValue;
+        const yValue = getProbabilityForDuration(d as Item[], startValue);
+        d3.select(d3.event.currentTarget)
+            .select('.area-clip-rect')
+            .attr('width', xScale(startValue));
+        d3.select(d3.event.currentTarget)
+            .select('.click-line-y')
+            .attr('y1', yScale(yValue))
+            .attr('x1', xScale(startValue))
+            .attr('x2', xScale(startValue))
+            .attr('opacity', 1);
+        d3.select(d3.event.currentTarget)
+            .select('.click-line-x')
+            .attr('y1', yScale(yValue))
+            .attr('y2', yScale(yValue))
+            .attr('x2', xScale(startValue))
+            .attr('opacity', 1);
+        d3.select(d3.event.currentTarget)
+            .select('.click-line-y-text')
+            .attr('x', xScale(startValue) - 40)
+            .attr('y', yScale(yValue) + 20)
+            .text(`${this.data.filter(c => c.duration <= startValue).length}/${
+                this.data.length}`)
+            .attr('opacity', 1);
+        d3.select(d3.event.currentTarget)
+            .select('.click-line-x-text')
+            .attr('x', xScale(startValue) / 2)
+            .attr('y', yScale(yValue) + 10)
+            .text(`${
+                (getProbabilityForDuration(this.data, startValue) * 100)
+                    .toFixed(1)}%`)
+            .attr('opacity', 1);
+        d3.select(d3.event.currentTarget)
+            .selectAll('.dots')
+            .data(this.data)
+            .attr(
+                'fill',
+                (dp: Item) => dp.duration <= startValue ? 'grey' : 'black');
+      }
+    });
+
+    // hover
+    const highlightColor = '#b9edc4';
+    const strokeColor = '#167364';
+    const circleColor = '#a0aade';
+    const labelsize = [40, 25];
```

Comment:
Why using a list instead of two variables? `[0]` and `[1]` are not very descriptive way.